tests/State/HubAdapterTest.php

```diff
@@ -105,6 +105,23 @@ public function testWithScope(): void
         HubAdapter::getInstance()->withScope($callback);
     }
 
+    public function testWithScopeWithReturnValue(): void
```

Comment:
As written above, this test can be squashed with the previous one and you can just check that the value returned by the callback is the one you're expecting